addons/zylann.scatter/tools/plugin.gd

```diff
@@ -277,21 +278,26 @@ static func get_scatter_child_instance(node, scatter_root):
 	return null
 
 
-func set_pattern(pattern):
-	if _pattern != pattern:
-		_pattern = pattern
-		var temp = pattern.instance()
-		# TODO This causes errors because of accessing `global_transform` outside the tree... Oo
-		# See https://github.com/godotengine/godot/issues/30445
-		var aabb = Util.get_scene_aabb(temp)
-		_pattern_margin = aabb.size.length() * 0.4
-		temp.free()
+func set_patterns(patterns):
+	if _patterns != patterns:
+		_patterns = patterns
```

Comment:
I wish GDScript was as precise as C++ when it comes to access specifiers, but this array comes from the node, and here we end up storing a reference to it, completely implicitely. This makes it easy for bugs to get introduced in the future if someone else decides to modify this array (at least, my coding style tries to avoid referencing internals in this way).

As mentionned earlier, I also think `set_patterns` may change to something else. The goal of the function originally was to update the placement margin, and cache the selected pattern:
- Your code now calculates the maximum AABB over all patterns, which is not valid. The margin should be calculated from selected patterns only.
- Caching all patterns might not be really necessary, since we can get that list anytime by using `_node.get_patterns()` anytime.

Reply:
Instead of getting the patterns directly, I used a `duplicated` version.